src/Nethermind/Nethermind.TxPool/TxPool.cs

```diff
@@ -459,17 +455,6 @@ public bool RemoveTransaction(Keccak? hash)
                     return false;
                 if (hasBeenRemoved)
                 {
-                    Address? address = transaction.SenderAddress;
-
-                    if (address != null && _nonces.TryGetValue(address, out AddressNonces? addressNonces))
-                    {
-                        addressNonces.Nonces.TryRemove(transaction.Nonce, out _);
-                        if (addressNonces.Nonces.IsEmpty)
-                        {
-                            _nonces.Remove(address, out _);
```

Comment:
Why is this not needed now? We are not removing nonces of txs removed from tx pool any more? What if we remove tx from pool and then receive it again in the future - it will not be rejected because of nonce already used?
Would be good to add test for:
1) receiving tx A from the network (not own tx)
2) receiving more txs with higher gas price, tx A is evicted from the pool
3) some txs are removed from the pool (e.g. included in the block), TxPool is not full now
4) receiving tx A from the network again (exactly the same as before) - tx A should be added to the pool

Reply:
NonceManager can't cause any tx rejection. All this logic is only for own transactions. And I implemented the test

Reply:
Makes sense. I have one more scenario. What if
1) receiving txA from senderA with nonce X from the network
2) receiving unsigned txB (without nonce) from senderA from RPC endpoint
I think in this scenario txA should have nonce X and txB should have nonce X+1. How it works in current implementation? Wouldn't both txs have nonce X?

Reply:
And ofc in this scenario txA would be in pending, not included in the block yet

Reply:
This test will fail, NonceManager receives information only from sendTx calls and from AccountStateProvider. I can add this feature, but I prefer not to for three reasons.
1. It will require storing all nonce of all accounts.
2. We will have to add NonceManager field into txPool. Not very bad, but we should avoid this.
3. It's a kinda not good approach to send transactions from multiple nodes at the same time.

Reply:
> It's a kinda not good approach to send transactions from multiple nodes at the same time.

I'm not 100% sure about this one, but ok, maybe you are right. Probably not worth to add a lot of complexity to cover this edge case scenario